lib/core/src/display/navigation/camera_manager.rs

```diff
@@ -0,0 +1,197 @@
+use crate::prelude::*;
+
+use super::mouse_manager::State;
+use super::mouse_manager::MouseManager;
+use super::mouse_manager::MouseClickEvent;
+use super::mouse_manager::MouseWheelEvent;
+use super::mouse_manager::MousePositionEvent;
+use super::mouse_manager::MouseButton;
+use crate::system::web::Result;
+use crate::display::rendering::DOMContainer;
+
+use nalgebra::Vector2;
+use std::rc::Rc;
+use std::cell::RefCell;
+use nalgebra::zero;
+
+// ====================
+// === ZoomingEvent ===
+// ====================
+
+/// A struct holding zooming event information.
+pub struct ZoomingEvent {
+    pub focus  : Vector2<f32>,
+    pub amount : f32
+}
+
+impl ZoomingEvent {
+    fn from_mouse_wheel(event:MouseWheelEvent, data:&Rc<RefCell<CameraManagerData>>) -> Self {
+        let amount = event.movement_y;
+        let focus  = data.borrow().mouse_position;
+        Self { amount, focus }
+    }
+
+    fn from_mouse_move(event:MousePositionEvent, focus:Vector2<f32>) -> Self {
+        let amount = event.position.y - event.previous_position.y;
+        Self { focus, amount }
+    }
+}
+
+// ====================
+// === PanningEvent ===
+// ====================
+
+/// A struct holding panning event information.
+pub struct PanningEvent {
+    pub movement : Vector2<f32>
+}
+
+impl PanningEvent {
+    fn from(event:MousePositionEvent) -> Self {
+        let mut movement   = event.position - event.previous_position;
+                movement.x = -movement.x;
+        Self { movement }
+    }
+}
+
+// =============
+// === Event ===
+// =============
+
+/// An enumeration representing navigation events.
+pub enum Event {
```

Comment:
This name is misleading. I see it in the code below and I'm thinking "Oh this is a generic event, because its called Event", but its not. Its a very specific kind of event.